maps/chronosphere/main.lua

```diff
@@ -1,565 +1,648 @@
+    game.surfaces[objective.active_surface_index].pollute(pos, exterior_pollution)
+    game.pollution_statistics.on_flow("locomotive", exterior_pollution)
+
+	::continue::
 end
 
-local function on_pre_player_left_game(event)
-    local objective = Chrono_table.get_table()
-    local player = game.players[event.player_index]
-    if player.controller_type == defines.controllers.editor then
-        player.toggle_map_editor()
-    end
-    if player.character then
-        objective.offline_players[#objective.offline_players + 1] = {index = event.player_index, tick = game.tick}
-    end
+local tick_minute_functions = {
+
+	[300 * 2] = Ai.destroy_inactive_biters,
+	[300 * 3 + 30 * 0] = Ai.pre_main_attack,		-- setup for main_attack
+	[300 * 3 + 30 * 1] = Ai.perform_main_attack,
+	[300 * 3 + 30 * 2] = Ai.perform_main_attack,
+	[300 * 3 + 30 * 3] = Ai.perform_main_attack,
+	[300 * 3 + 30 * 4] = Ai.perform_main_attack,
+	[300 * 3 + 30 * 5] = Ai.perform_main_attack,	-- call perform_main_attack 7 times on different ticks
+	[300 * 4] = Ai.send_near_biters_to_objective,
+	[300 * 5] = Ai.wake_up_sleepy_groups,
+	--[300] = Ai.rogue_group
+
+}
+
+local function initiate_jump_countdown()
+	local objective = Chrono_table.get_table()
+	local difficulty = Difficulty.get().difficulty_vote_value
+	local length = Balance.generate_jump_countdown_length(difficulty)
+
+	objective.jump_countdown_start_time = objective.passivetimer
+	objective.jump_countdown_length = length
+	game.print({"chronosphere.message_initiate_jump_countdown", length}, {r=0.98, g=0.66, b=0.22})
 end
 
-local function set_objective_health(final_damage_amount)
-    if final_damage_amount == 0 then
-        return
-    end
-    local objective = Chrono_table.get_table()
-    objective.health = math_floor(objective.health - final_damage_amount)
-    if objective.health > objective.max_health then
-        objective.health = objective.max_health
-    end
+local function check_if_overstayed()
+	local objective = Chrono_table.get_table()
 
-    if objective.health <= 0 then
-        Chrono.objective_died()
-    end
-    if objective.health < objective.max_health / 2 and final_damage_amount > 0 then
-        Upgrades.trigger_poison()
-    end
-    rendering.set_text(objective.health_text, 'HP: ' .. objective.health .. ' / ' .. objective.max_health)
+	  if objective.passivetimer * objective.passive_chronocharge_rate > (objective.chronochargesneeded * 0.75) and objective.chronojumps >= Balance.jumps_until_overstay_is_on(Difficulty.get().difficulty_vote_value) then
+		  objective.overstaycount = objective.overstaycount + 1
+	  end
 end
 
-function Public.chronojump(choice)
-    local Diff = Difficulty.get()
-    local objective = Chrono_table.get_table()
-    if objective.game_lost then
-        goto continue
-    end
-    Chrono.process_jump()
+function Public.add_chronocharge()

```

Comment:
Late review comment.  I'm a lua newb but it doesn't seem like this function does anything, possibly an unfinished or leftover change?